src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -739,6 +741,76 @@ class StringGenerator : public Generator<std::string> {
     const size_t _alphabetLength;
 };
 
+/** `{^ChooseFromDataset:{...}` */
+class RandomStringFromDataset : public Generator<std::string> {
+
+public:
+    RandomStringFromDataset(const Node& node, GeneratorArgs generatorArgs)
+        : _rng{generatorArgs.rng},
+          _id{generatorArgs.actorId},
+          _path{node["path"].maybe<std::string>().value()},
+          _pathLength{_path.size()}{}
+
+    void loadDataset() {
+      {
+        if (_pathLength <= 0) {
+            BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax(
+            "ChooseFromDataset requieres non-empty path"));
+            }
+
+        {
+            if (_all_datasets.count(_path) == 0){
+                _ifs.open(_path, std::ifstream::in);
+                if (_ifs.is_open()){
+                    while(std::getline(_ifs, _current_line))
+                    {
+                        // Ignore emtpy lines
+                        if (_current_line.size()!=0){
+                            _all_datasets[_path].push_back(_current_line);
+                        }
+                    }
+                }
+            
+                else {
+                    BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax(
+                        "The specified file for ChooseFromDataset cannot be opened or it does not exist")); }
+
+                if (_all_datasets[_path].size() == 0) {
+                    BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax(
+                        "The specified file for ChooseFromDataset is empty")); }
+            }
+        }
+      }
+    }
+
+    std::string evaluate() {
```

Comment:
This means the first invocation of the generator will (presumably) be orders of magnitude slower than subsequent invocations. Can you prime this at ctor-time?

Reply:
Based on a comment from David I moved the loadDataset to the constructor instead of being part validate().